hw/xbox/nv2a/pgraph.c

```diff
@@ -5554,6 +5555,30 @@ static void surface_copy_expand(uint8_t *out, uint8_t *in, unsigned int width,
     }
 }
 
+void export_pgraph_data(NV2AState *d, SurfaceBinding *surface)
+{
+    //FIXME: correct memory allocation for imagedata.
+    //FIXME: Export correctly data from pg and Surface into ExportData
+    PGRAPHState *pg = &d->pgraph;
+    ExportData *imagedata = (ExportData*)malloc(pg->surface_shape.clip_x * pg->surface_shape.clip_y * pg->surface_shape.clip_width * pg->surface_shape.clip_height 
```

Comment:
This is totally wrong, you're allocating a potentially huge amount of memory by doing all this multiplication. From the context, you're just trying to allocate an ExportData instance, which should be a simple `sizeof(ExportData)` malloc.